File: base/src/main/java/io/quarkus/code/config/PlatformConfig.java
package io.quarkus.code.config;

import io.smallrye.config.ConfigMapping;
import io.smallrye.config.WithName;

import java.util.Optional;

@ConfigMapping(prefix = "io.quarkus.code.quarkus-platforms")
public interface PlatformConfig {

    @WithName("reload-cron-expr")
    String getReloadCronExpr();

    @WithName("registry-id")
    Optional<String> getRegistryId();
}

File: api/src/main/kotlin/io/quarkus/code/config/PlatformConfig.kt
package io.quarkus.code.config

import io.smallrye.config.ConfigMapping
import io.smallrye.config.WithName
import java.util.*

@ConfigMapping(prefix = "io.quarkus.code.quarkus-platforms")
interface PlatformConfig {

    @get:WithName("reload-cron-expr")
    val reloadCronExpr: String

    @get:WithName("registry-id")
    val registryId: Optional<String>
}
